projects/sound-open-firmware/Dockerfile

```
@@ -21,21 +21,25 @@
 FROM gcr.io/oss-fuzz-base/base-builder
 
 # Base packages
-RUN apt-get -y update
+RUN apt-get update
 RUN apt-get install -y \
       gettext git libc6-dev-i386 libglib2.0-dev libncurses5-dev \
       libtool ninja-build python3-pip
 RUN pip3 install west
 
 # Zephyr SDK:
 #
-# Zephyr doesn't provide a clean "get the latest version" URL, but
-# note that the use of the /latest/ path component at least ensures
-# this will fail on a new release.
-ARG SDK_VER=0.16.1
+# Zephyr doesn't provide a clean "get the latest version" URL, so we
+# get it by cloning the sdk-ng tree (it's 15% the size of the binary
+# we'll grab, so minimal overhead) and finding the last tag.
 WORKDIR /root
-RUN curl -L -o sdktmp.tar.xz https://github.com/zephyrproject-rtos/sdk-ng/releases/latest/download/zephyr-sdk-${SDK_VER}_linux-x86_64_minimal.tar.xz
-RUN tar xf sdktmp.tar.xz; rm sdktmp.tar.xz
+RUN git clone https://github.com/zephyrproject-rtos/sdk-ng
+RUN cd sdk-ng; git tag | grep -P '^v[0-9\.]+$' | sort -n | tail -1 | sed 's/v//' > ../SDKVER
+RUN sh -c "echo https://github.com/zephyrproject-rtos/sdk-ng/releases/download/v`cat SDKVER`/zephyr-sdk-`cat SDKVER`_linux-x86_64_minimal.tar.xz > SDKURL"
```

What are these `sh -c` for? They increase the level of nesting for no obvious reason. `RUN` clearly involves a shell=True already, otherwise your `git tag` above wouldn't work either.

backticks are basically deprecated https://www.shellcheck.net/wiki/SC2006

```suggestion
RUN echo "https://github.com/zephyrproject-rtos/sdk-ng/releases/download/v$(cat SDKVER)/zephyr-sdk-$(cat SDKVER)_linux-x86_64_minimal.tar.xz" > SDKURL
```

---

Couldn't figure out how to get a computed value into a variable for a subsequent RUN command, and there seemed to be some voodoo around the `$` being interpreted at the Dockerfile level and not the shell, thus the old syntax.  I'm absolutely not the person to ask about proper dockerfile form, but for sure this is deeper than mere bourne shell propriety.